src/contracts/core/StakeRootCompendium.sol

```diff
@@ -234,226 +202,155 @@ contract StakeRootCompendium is StakeRootCompendiumStorage {
+    }
+
+    function _updateCumulativeCharges() internal {
+        // todo comment on type casting
+        (uint96 _totalChargeLinear, uint96 _totalChargeConstant) = _cumulativeCharges();
+        if (_totalChargeLinear != totalChargeLinear || _totalChargeConstant != totalChargeConstant) {
+            totalChargeLinear = _totalChargeLinear;
+            totalChargeConstant = _totalChargeConstant;
+            blockNumberLastChargeUpdated = uint64(block.number);
         }
+    }
 
-        return penalty;
+    function _cumulativeCharges() internal view returns (uint96, uint96) {
+        uint96 proofEpochs = uint96((block.number - blockNumberLastChargeUpdated) / proofInterval);
+        return (totalChargeLinear + chargePerProofLinear * proofEpochs, totalChargeConstant + chargePerProofConstant * proofEpochs);
     }
 
-    // updates total strategies and the total charge per proof whenever an operator set's number of strategies changes
-    function _updateTotals(uint256 numStrategiesBefore, uint256 numStrategiesAfter) internal {
-        totalStrategies = totalStrategies - numStrategiesBefore + numStrategiesAfter;
-        totalChargeSnapshot.push(
-            uint32(block.timestamp), 
-            uint224(totalStrategies * linearChargePerProof + constantChargePerProof)
-        );
+    // updates the deposit balance for the operator set and returns the penalty if the operator set has fallen below the minimum deposit balance
+    function _updateDepositInfo(IAVSDirectory.OperatorSet memory operatorSet) internal returns (uint256) {
+        uint256 depositBalance = getDepositBalance(operatorSet);
+        depositInfo[operatorSet.avs][operatorSet.operatorSetId].balance = uint96(depositBalance);
+        depositInfo[operatorSet.avs][operatorSet.operatorSetId].snapshotTimestamp = uint64(block.timestamp);
+        return depositBalance;
     }
 
     function _postStakeRoot(
-        uint32 calculationTimestamp,
-        bytes32 stakeRoot,
-        address chargeRecipient,
-        bool forcePosted
+        bool isProofPost,
+        uint256 calculationBlockNumber,
+        bytes32 stakeRoot
     ) internal {
-        require(
-            calculationTimestamp % proofInterval == 0,
-            "StakeRootCompendium._postStakeRoot: calculationTimestamp must be a multiple of proofInterval"
-        );
+        StakeRootSubmission memory submission = stakeRootSubmissions[calculationBlockNumber];
 
-        uint256 stakeRootSubmissionsLength = stakeRootSubmissions.length;
-        if (stakeRootSubmissionsLength != 0) {
-            require(
-                stakeRootSubmissions[stakeRootSubmissionsLength - 1].calculationTimestamp + proofInterval == calculationTimestamp,
-                "StakeRootCompendium._postStakeRoot: calculationTimestamp must be greater than the last posted calculationTimestamp"
-            );
+        if (submission.stakeRoot == bytes32(0)) {
+            submission.stakeRoot = stakeRoot;
```

Comment:
if this is not a proof post, the submission must be proven. confirmation must happen after proofs

Reply:
I was kind of overloading this functionality for the desired feature that if the provers are offline, the same data structures can be used for AVS' which may choose to use a simpler confirmation rule (trust eigen labs)

---

or what is desired as a fallback if zkprovers are all offline?

Reply:
but yeah, certainly can adjust this so confirmerss strictly confirm